atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -84,13 +84,13 @@ public class CassandraService implements AutoCloseable {
     private final CassandraClientPoolMetrics poolMetrics;
     private final Refreshable<CassandraKeyValueServiceRuntimeConfig> runtimeConfig;
 
-    private volatile RangeMap<LightweightOppToken, List<CassandraServer>> tokenMap = ImmutableRangeMap.of();
+    private volatile RangeMap<LightweightOppToken, ImmutableSet<CassandraServer>> tokenMap = ImmutableRangeMap.of();
     private final Map<CassandraServer, CassandraClientPoolingContainer> currentPools = new ConcurrentHashMap<>();
     private volatile Map<CassandraServer, String> hostToDatacenter = ImmutableMap.of();
 
     private List<CassandraServer> cassandraHosts;
 
-    private volatile Set<CassandraServer> localHosts = ImmutableSet.of();
+    private volatile ImmutableSet<CassandraServer> localHosts = ImmutableSet.of();
```

Comment:
ensure we're always reasoning about an immutable snapshot copy of local Cassandra hosts that is refreshing asynchronously